indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -365,6 +365,7 @@ public TaskStatus call()
                                   )
                               );
 
+                              command.add(String.format("-Ddruid.indexer.taskId=%s", task.getId()));
```

Comment:
Basically you should be able to inject `io.druid.server.metrics.DataSourceTaskIdHolder` and get either `null` or the task ID from that, depending on if the instance even makes sense to have such a thing.